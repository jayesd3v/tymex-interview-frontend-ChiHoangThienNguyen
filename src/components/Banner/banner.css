.banner {
    background: url(/public/banner-background.png);
    background-repeat: no-repeat;
    background-position: bottom center;
    background-size: cover;
    position: relative;
}

.banner::before {
    content: '';
    position: absolute;
    background: #000000b3;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}

.banner > .new-arrival {
    position: absolute;
    width: 60%;
    left: 10%;
    top: 25%;
}

.banner > .new-arrival > img {
    width: 100%;
}

.banner > .banner-holder {
    background: url(/public/banner-holder.png);
    background-repeat: no-repeat;
    background-size: cover;
    height: 230px;
    width: 100%;
    position: absolute;
    bottom: 0;
    left: 0;
}

.mobile-scrollable-section {
    display: flex;
    position: relative;
    overflow-x: scroll;
    overflow-y: hidden;
}