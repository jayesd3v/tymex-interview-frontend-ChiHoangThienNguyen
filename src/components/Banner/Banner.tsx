import { Container } from 'react-bootstrap';
import NewCharacterTile from '../NewCharacterTile';
import './banner.css';
import clsx from 'clsx';
import { NewCharacterTileProps } from '../NewCharacterTile/NewcharacterTile';

const NEW_CHARACTERS: NewCharacterTileProps[] = [
    {
        character: 'thedj',
        name: 'THE DJ',
    },
    {
        character: 'assassin',
        name: 'ASSASSIN',
    },
    {
        character: 'basketball-girl',
        name: 'BASKETBALL GIRL',
    },
    {
        character: 'mafia-england',
        name: 'MAFIA ENGLAND',
    },
    {
        character: 'neon-guy',
        name: 'NEON GUY',
    },
];

const Banner = () => {
    const renderNewCharacters = () =>
        NEW_CHARACTERS.length && (
            <div className='mobile-scrollable-section pt-5 pb-5 ps-5'>
                <div className='d-flex'>
                    {NEW_CHARACTERS.map(({ character, name }: NewCharacterTileProps, index) => (
                        <NewCharacterTile
                            key={character}
                            character={character}
                            name={name}
                            className={clsx(index === 0 && 'd-lg-none')}
                        />
                    ))}
                </div>
            </div>
        );

    const renderSpotlightCharacter = () => {
        const { character } = NEW_CHARACTERS[0];
        return (
            <div className='d-none d-lg-block spotlight-character'>
                <div className='image' style={{ backgroundImage: `url(/${character}.png)` }}></div>
            </div>
        );
    };

    return (
        <div className='banner'>
            <div className='new-arrival'>
                <img src='/new-arrival.png' alt='new-arrival' />
            </div>
            <div className='banner-holder'>
                <Container className='pt-3 pt-xl-3'>
                    {renderNewCharacters()}
                    {renderSpotlightCharacter()}
                </Container>
            </div>
        </div>
    );
};

export default Banner;
