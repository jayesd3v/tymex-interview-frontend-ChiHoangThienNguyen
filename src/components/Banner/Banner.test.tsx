import { render } from "@testing-library/react";
import Banner from "./Banner";

describe('Banner tests', () => {
    test('snapshot test', () => {
        const { container } = render(<Banner />);
        expect(container).toMatchSnapshot();
    });
});