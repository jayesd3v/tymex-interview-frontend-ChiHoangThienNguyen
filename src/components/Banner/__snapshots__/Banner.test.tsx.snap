// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner tests snapshot test 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="new-arrival"
    >
      <img
        alt="new-arrival"
        src="/new-arrival.png"
      />
    </div>
    <div
      class="banner-holder"
    >
      <div
        class="pt-3 pt-xl-3 container"
      >
        <div
          class="mobile-scrollable-section pt-5 pb-5 ps-5"
        >
          <div
            class="d-flex"
          >
            <div
              class="new-character-tile me-5 d-lg-none"
            >
              <div
                class="character thedj"
              />
              <div
                class="name text-center"
              >
                THE DJ
              </div>
            </div>
            <div
              class="new-character-tile me-5"
            >
              <div
                class="character assassin"
              />
              <div
                class="name text-center"
              >
                ASSASSIN
              </div>
            </div>
            <div
              class="new-character-tile me-5"
            >
              <div
                class="character basketball-girl"
              />
              <div
                class="name text-center"
              >
                BASKETBALL GIRL
              </div>
            </div>
            <div
              class="new-character-tile me-5"
            >
              <div
                class="character mafia-england"
              />
              <div
                class="name text-center"
              >
                MAFIA ENGLAND
              </div>
            </div>
            <div
              class="new-character-tile me-5"
            >
              <div
                class="character neon-guy"
              />
              <div
                class="name text-center"
              >
                NEON GUY
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-none d-lg-block spotlight-character"
        >
          <div
            class="image"
            style="background-image: url(/thedj.png);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
